From: Michal Hocko <mhocko@suse.cz>
Subject: igb: don't reuse pages with pfmemalloc flag fix
Git-commit: 2f064f3485cd29633ad1b3cfb00cc519509a3d72 (partial)
References: bnc#920016
Patch-mainline: 4.2-rc8

Partial backport, the rest is in
patches.fixes/mm-make-page-pfmemalloc-check-more-robust.patch much earlier in
the series. We want to keep the patch close to the one which has introduced
page::pfmemalloc so this follow up.

The fix is really trivial page->pfmemalloc needs to be checked via
page_is_pfmemalloc(page) now.

Signed-off-by: Michal Hocko <mhocko@suse.cz>

---
 drivers/net/ethernet/intel/igb/igb_main.c |    4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

--- a/drivers/net/ethernet/intel/igb/igb_main.c
+++ b/drivers/net/ethernet/intel/igb/igb_main.c
@@ -6423,7 +6423,7 @@ static bool igb_can_reuse_rx_page(struct
 	if (unlikely(page_to_nid(page) != numa_node_id()))
 		return false;
 
-	if (unlikely(page->pfmemalloc))
+	if (unlikely(page_is_pfmemalloc(page)))
 		return false;
 
 #if (PAGE_SIZE < 8192)
@@ -6494,7 +6494,7 @@ static bool igb_add_rx_frag(struct igb_r
 
 		/* we can reuse buffer as-is, just make sure it is local */
 		if (likely((page_to_nid(page) == numa_node_id()) &&
-			   !page->pfmemalloc))
+			   !page_is_pfmemalloc(page)))
 			return true;
 
 		/* this page cannot be reused so discard it */
